src/replica_server/replication_lib/replica_init.cpp

```diff
@@ -269,9 +269,8 @@ error_code replica::init_app_and_prepare_list(bool create_new)
                         _private_log->max_commit_on_disk(),
                         finish_time - start_time
                         );
+
                     _private_log->check_valid_start_offset(get_gpid(), _app->init_info().init_offset_in_private_log);
-                    set_inactive_state_transient(true);
```

Comment:
Here we set the value according to per-replica state, and in replica-stub we set that for shared state. It is best to leave both, similar to the case where we set last error multiple times in different scope for different scenarios.